File: backend/app/schemas/api_key.py
from datetime import datetime
from typing import Optional
import uuid

from pydantic import BaseModel, Field
from dataclasses import dataclass


class ApiKeyCreate(BaseModel):
    tenant_id: uuid.UUID
    channel_id: Optional[uuid.UUID] = None
    name: str = Field(..., min_length=1, max_length=150)
    expires_at: Optional[datetime] = None


class ApiKeyResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    channel_id: Optional[uuid.UUID] = None
    name: str
    key_prefix: str
    is_active: bool
    created_at: datetime
    last_used_at: Optional[datetime] = None
    expires_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class ApiKeyCreatedResponse(ApiKeyResponse):
    api_key: str


@dataclass
class MachineIdentity:
    api_key_id: uuid.UUID
    tenant_id: uuid.UUID
    channel_id: Optional[uuid.UUID]
    name: str


@dataclass
class WebhookAuthContext:
    api_key: MachineIdentity
    tenant_id: uuid.UUID
    channel_id: uuid.UUID
    provider_signature: Optional[str] = None

File: frontend/src/types/apiKey.ts
export interface ApiKeyItem {
  id: string;
  tenant_id: string;
  channel_id: string | null;
  name: string;
  key_prefix: string;
  is_active: boolean;
  created_at: string;
  last_used_at: string | null;
  expires_at: string | null;
}

export interface ApiKeyCreatePayload {
  tenant_id: string;
  channel_id?: string;
  name: string;
  expires_at?: string;
}

export interface ApiKeyCreatedItem extends ApiKeyItem {
  api_key: string;
}
